compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/SyntheticAccessorLowering.kt

```diff
@@ -387,13 +417,23 @@ private class SyntheticAccessorLowering(val context: JvmBackendContext) : IrElem
             return false
 
         val contextDeclarationContainer = allScopes.lastOrNull { it.irElement is IrDeclarationContainer }?.irElement
+        // context for functions in an interface is DefaultImpls
+        val adjustedContext = contextDeclarationContainer.let {
```

Comment:
> Any super call from an interface should be a static call to the super-interface's DefaultImpls by this point.

May I know how to determine if a call is static or not? I tried to check whether it has a dispatch receiver as well as IrFunction.isStatic (almost the same), but with [simpleCallWithHierarchy](https://github.com/JetBrains/kotlin/blob/master/compiler/testData/compileKotlinAgainstKotlin/jvm8/jvm8against6/simpleCallWithHierarchy.kt), the callee `Test.test` doesn't look like static but is generated with invokestatic. In that example, without adjusting the context, it would generate an unnecessary accessor.

Reply:
1. A static function is indeed one that is `isStatic` (it should not have a `dispatchReceiverParameter`), and a static call is one without a `dispatchReceiver`. What exceptions do you have?
2. The example in question does not have an `INVOKESPECIAL`. In fact, `INVOKESTATIC` was generated there only because the current descriptor-based type mapper is clever enough to recognize interface super calls. A better way is to lower them at an earlier stage. Here is a branch that does exactly that: https://github.com/JetBrains/kotlin/tree/rr/gogabr/interface-super-calls

Reply:
@gogabr Thanks, that's what I'm looking for. I've been trying to adjust the logic according to typemapper's behavior. Apparently I was in a wrong direction.

I'll undo the context adjustment and wait for your patch to land. Please correct me if I misunderstand anything.

Reply:
@gogabr Thanks for the branch! I've tested the updated PR with your branch and it works well.

Reply:
That PR is now merged into master.